### PR TITLE
Adds diagrams and updates doc

### PR DESCRIPTION
**Pull request by nkcr**

### CODE REVIEW

**Comment by nkcr** [comment]
Since GitHub doesn't display images in PR files, here are the diagrams:

## Overview

overview

## Stacks

stacks

## Dela flow

dela flow

## Ordering flow

cosipbft

## (for reference) Package deps

packages

File: docs/coverpage.md [hunk 1]
```diff
@@ -12,6 +12,6 @@
 </p>
 
 - A set of modular abstractions to perform DL operations
-- Provides an implementation of a minimalistic DL
-- Demonstrates a full working DL implementation based on [Byzcoin](https://www.usenix.org/system/files/conference/usenixsecurity16/sec16_paper_kokoris-kogias.pdf)
+- Provides an implementation of a minimalist DL
+- Demonstrates a full working DL implementation based on [Collective Signing](https://www.usenix.org/system/files/conference/usenixsecurity16/sec16_paper_kokoris-kogias.pdf)
 - Offers a platform that supports research in DLT
```
Comment (Gilthoniel) [inline docs/coverpage.md:17]: Like David mentioned, it is not specified what DLT means.

File: docs/introduction.md [hunk 2]
```diff
@@ -46,65 +46,22 @@ crypto-currency.
...
-## Complete packages dependencies
-
-This is still a WIP, here is the current interactions between the packages:
-
-![](assets/packages.png)
+Dela has 6 core modules that you will find in `core/`:
+
+- access
+- execution
+- ordering (and its sub-module ordering/pool)
```
Comment (Gilthoniel) [inline docs/introduction.md:53]: `txn/pool` you mean ?

**Review by Gilthoniel — APPROVED** [review]

File: docs/architecture.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Architecture
+
+Dela has been built with modularization and testability in mind. As such, the
+system is made of multiple interoperable modules' implementations based on a set
```
Comment (pierluca) [inline docs/architecture.md:4]:
```suggestion
system is made of multiple interoperable modules based on a set
```

File: docs/architecture.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+of well-defined abstractions.
+
+## Stacks
+
+The stacks display the role of each module and its relation to others. Each
+module can rely on zero or more of the modules bellow it. From the client's
```
Comment (pierluca) [inline docs/architecture.md:10]:
```suggestion
module can rely on zero or more of the modules below it. From the client's
```

File: docs/architecture.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+perspective the stack is simpler since it doesn't need to know about the complex
+ordering mechanism.
+
+![stacks](assets/stacks.png)
+
+## Execution flow
+
+The execution flow displays how the system uses the modules in order to fullfil
```
Comment (pierluca) [inline docs/architecture.md:18]:
```suggestion
The execution flow displays how the system uses the modules in order to fulfill
```

**Review by pierluca — COMMENTED** [review]
Minor changes, I'll submit the feedback on images separately.

**Comment by pierluca** [comment]
cosipbft.png --> I don't see something coming out of the "validation" block. Is this correct? I would assume that once we call validate(), there's some sort of return value or data flow. Is the arrow between `pbft` and `validation` in the right direction? Should it be bidirectional?

**Comment by nkcr** [comment]
> Is the arrow between `pbft` and `validation` in the right direction? Should it be bidirectional?

The validation actually returns an error or nil, and, based on that, pbft commits the staging tree or not

**Comment by pierluca** [comment]
> > Is the arrow between `pbft` and `validation` in the right direction? Should it be bidirectional?
> 
> The validation actually returns an error or nil, and, based on that, pbft commits the staging tree or not

Should we then add that "return" arrow in the diagram ? Am I missing something? :)

**Comment by nkcr** [comment]
> Should we then add that "return" arrow in the diagram ? Am I missing something? :)

I don't want to confuse with a return. The point is that the stageTree is passed as reference, and modified "in place" by the validation/execution service. I'm updating the diagrams to reflect that.

**Review by pierluca — APPROVED** [review]
Awesome :) LGTM

**Pull request merged**